kernel/poll.c

```diff
@@ -76,9 +76,14 @@ static inline bool is_condition_met(struct k_poll_event *event, u32_t *state)
 		break;
 	case K_POLL_TYPE_IGNORE:
 		break;
+	/* LCOV_EXCL_START
```

Comment:
This is not the right approach.
We need to change these asserts to runtime errors, and propagate these errors to the caller.